Sources/Identity/CustomerInfoManager.swift

```diff
@@ -90,27 +90,67 @@ class CustomerInfoManager {
+            }
+
+        case .cachedOrFetched:
+            let infoFromCache = self.cachedCustomerInfo(appUserID: appUserID)
+            var completionCalled = false
 
-        if let infoFromCache = infoFromCache {
-            Logger.debug(Strings.customerInfo.vending_cache)
-            if let completion = completion {
-                completionCalled = true
-                operationDispatcher.dispatchOnMainThread {
-                    completion(.success(infoFromCache))
+            if let infoFromCache = infoFromCache {
+                Logger.debug(Strings.customerInfo.vending_cache)
+                if let completion = completion {
+                    completionCalled = true
+                    self.operationDispatcher.dispatchOnMainThread {
+                        completion(.success(infoFromCache))
+                    }
                 }
             }
-        }
 
-        // Prevent calling completion twice.
-        let completionIfNotCalledAlready = completionCalled ? nil : completion
+            // Prevent calling completion twice.
+            let completionIfNotCalledAlready = completionCalled ? nil : completion
 
-        systemInfo.isApplicationBackgrounded { isAppBackgrounded in
-            self.fetchAndCacheCustomerInfoIfStale(appUserID: appUserID,
-                                                  isAppBackgrounded: isAppBackgrounded,
-                                                  completion: completionIfNotCalledAlready)
+            self.systemInfo.isApplicationBackgrounded { isAppBackgrounded in
+                self.fetchAndCacheCustomerInfoIfStale(appUserID: appUserID,
+                                                      isAppBackgrounded: isAppBackgrounded,
+                                                      completion: completionIfNotCalledAlready)
+            }
+
+        case .notStaleCachedOrFetched:
+            let infoFromCache = self.cachedCustomerInfo(appUserID: appUserID)
+
+            self.systemInfo.isApplicationBackgrounded { isAppBackgrounded in
+                let isCacheStale = self.deviceCache.isCustomerInfoCacheStale(appUserID: appUserID,
+                                                                             isAppBackgrounded: isAppBackgrounded)
+
+                if let infoFromCache = infoFromCache, !isCacheStale {
+                    Logger.debug(Strings.customerInfo.vending_cache)
+                    if let completion = completion {
+                        self.operationDispatcher.dispatchOnMainThread {
+                            completion(.success(infoFromCache))
+                        }
+                    }
+                } else {
+                    self.fetchAndCacheCustomerInfo(appUserID: appUserID,
+                                                   isAppBackgrounded: isAppBackgrounded,
+                                                   completion: completion)
+                }
```

Comment:
This behavior makes a lot more sense. I wonder if we should change some calls to use this instead now.

Reply:
I agree. Let's run it by the support team, they might have thoughts based on customer feedback.

I think we can merge this PR as is (with the current behavior), then ask the support team (and broader team) for feedback regarding the proposed change, then if folks agree make a release that makes this the default. 
We'd have to also make a purchases-android equivalent release so that we don't have inconsistent behavior in the hybrids.

Reply:
Well, I can see your reasoning... I think this is ok as long as this option is not the default... But if we wanted to make this the default as was mentioned on this comment, we probably want to be more lenient and allow to use the stale data... I'm sure there are a ton of people that would use the apps while offline that would cause this to error out even if we had a cached value that the customer can potentially use.

Reply:
> Well, I can see your reasoning... I think this is ok as long as this option is not the default... But if we wanted to make this the default as was mentioned on this comment, we probably want to be more lenient and allow to use the stale data...

Totally agreed.

So we want this to be `notStaleCachedOrFetchedUnlessThatFailsInWhichCaseStaleCache`. I think this is pointing to the fact that really there's a lot of different combinations, and we can't offer them all. So maybe the right answer is to make sure this is documented correctly (could add a note to this policy).
To create that behavior users could do it themselves. I think it's best if we provide the primitives that users can compose around:
```swift
let customerInfo: CustomerInfo?

do {
  customerInfo = purchases.customerInfo(fetchPolicy: .notStaleCachedOrFetched)
} catch {
  customerInfo = purchases.customerInfo(fetchPolicy: .fromCacheOnly)
}
```

What do you guys think?

Reply:
Yeah, I think with the current options, users have a lot of flexibility, so as long as it's well documented, customers should be able to do what they need. However, as I mentioned, I wouldn't change the default to `notStaleCachedOrFetched` since I think that can potentially cause a bunch of issues for users.

We can talk about what the default behavior should be if we think the current behavior (`cachedOrFetched`) is causing issues